pkgs/top-level/all-packages.nix

```diff
@@ -7512,7 +7512,8 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) ApplicationServices;
   };
 
-  graphviz-nox = graphviz.override {
+  graphviz-nox = callPackage ../tools/graphics/graphviz {
+    inherit (darwin.apple_sdk.frameworks) ApplicationServices;
```

Comment:
Please change those back to override. Otherwise we need to maintain inherits at two places.

Reply:
That will cause infinite recursion

Reply:
Can you research where that happens? I would assume that if we are using less libraries then this should not happen?

Reply:
The overlay replaces graphviz with graphviz-nox  and graphviz-nox is graphviz overridden 

graphviz -> graphviz-nox --> graphviz

Reply:
@roberth also said the same thing 
![image](https://user-images.githubusercontent.com/56650223/205523876-5ee177fa-d717-427f-82dd-fc4906ded5b4.png)

Reply:
Oh, okay. Then we should add a very short comment so that we still know this later.